src/quickFixProvider.ts

```diff
@@ -20,7 +20,7 @@ class QuickFixProvider implements CodeActionProvider {
         }
 
         /* Fix all in selection */
-        if (!range.isEmpty && codeActions.length > 1) {
+        if (!window.activeTextEditor?.selection?.isEmpty) {
```

Comment:
@karlvr I found that the previous implementation does not work as expected when

Click the yellow bulb in the `PROBLEM` panel
![WeChat Screenshot_20210119161520](https://user-images.githubusercontent.com/6193897/105007084-82abc980-5a72-11eb-905a-73e7f3a302a1.png)

Click quick fix link in the diagnostic hover box
![WeChat Screenshot_20210119161540](https://user-images.githubusercontent.com/6193897/105007098-87707d80-5a72-11eb-86e0-b431e7253ba7.png)

So I changed to check if the current editor has selection or not.

Reply:
@jdneo interesting! You also removed the check for `codeActions.length > 1`. Was that intentional? That was intended not to show the "Fix all ... in selection" if there was only one option... but perhaps it's more consistent to always show it?

Reply:
@karlvr Good catch! I'll add that check back

Reply:
@karlvr Updated, do you have any more feedback about this PR?

Reply:
@jdneo I'm not sure about the active window check ... there's no guarantee that the active window is our document?

Reply:
@karlvr That's a good point. How about:

```
codeActions.length > 1 && document.uri === window.activeTextEditor?.document?.uri && !window.activeTextEditor?.selection?.isEmpty
```

Reply:
@jdneo actually I don't understand the original problem; is the range not reported correctly when coming from the quick fix in the problems panel?

Reply:
Let me attach a video to explain


https://user-images.githubusercontent.com/6193897/105280700-51073f80-5be5-11eb-98d0-87622acb3bf4.mp4

Reply:
@jdneo thank you, it's really interesting. Why isn't the `context.diagnostics` just the diagnostic that the user has clicked in the Problems tab? It seems to me that when I click on a lightbulb on a problem that it selects the context of the problem... thereby making a selection? Doesn't that mean that our doc would have a selection anyway when we go to check if the doc has a selection?

Reply:
I think the differences are:

- in the two way in the above video, there is a determined diagnostic, so the range refers to the diagnostic's range.
- when we click the yellow bulb in the editor (not in the Problem panel), it pops all the available diagnostics in this line, so now VS Code is not quite sure which diagnostic will be selected, so in that case, the range is empty.

Basically, the range here actually refers to the range of the diagnostics, not the selection range in the current editor.

Reply:
Does that make offering to fix the problems in the selection wrong?

Reply:
The main concern is it might confuse the users since there is no selection in the editor actually. Besides, it seems not necessary in that case to show the `fix in selection` option?